### DISCUSSION TITLE
DuckDB-less installation instructions conflict with pyproject.toml requirements

### DISCUSSION THREAD

**Discussion by boldorider4** [moj-analytical-services/splink#1826] [Q&A] [answered]
In [duckdbless_requirements.txt](https://github.com/moj-analytical-services/splink/blob/v3.9.10/scripts/duckdbless_requirements.txt) it is implied that `duckdb` is an optional dependency.

However in multiple modules in the project, `duckdb` is imported, also outside of the `splink/duckdb` subdirectory, which would make splink throw an exception at runtime.

Also, it is not clear in the [pyproject.toml](https://github.com/moj-analytical-services/splink/blob/v3.9.10/pyproject.toml) why duckdb is listed as a mandatory requirement, which will have `pip check` fail on the built package citing a missing `duckdb`.

**Comment by boldorider4** [comment]
To me it seems that the [duckdbless_requirements.txt](https://github.com/moj-analytical-services/splink/blob/v3.9.10/scripts/duckdbless_requirements.txt) were formulated by inspecting the installed package of an arbitrary duckdb-less environment with poetry. Which also poses the question whether the compatibility ranges of the [pyproject.toml](https://github.com/moj-analytical-services/splink/blob/v3.9.10/pyproject.toml) are reliable since they differ from the duckdb-less environment displayed.

**Reply by RobinL** [reply]
Not sure, some of the other team may be able to help with the details.  But at a quick look, the duckdb imports outside of the duckdb/ directory are within try/except blocks.

I'm reasonably confident the duckdb-less install works, because we have colleagues who rely on it

**Reply by boldorider4** [reply]
They are not all within try/except blocks, that's exactly my point.
see [here](https://github.com/moj-analytical-services/splink/blob/ac1cf23d41215492b68d56841c46de3bd93c7e99/splink/duckdb/duckdb_helpers/duckdb_helpers.py#L6)
 and [here](https://github.com/moj-analytical-services/splink/blob/ac1cf23d41215492b68d56841c46de3bd93c7e99/splink/comparison_helpers.py#L1).

**Reply by RobinL** [reply, marked as answer]
Sure, but have you tried to use the duckdbless install?  Those are a few ancillary functions which are not used by the main linking routines.  You're right , we should have been more careful about namespacing them but in practice I don't think it matters - you should be able to carry out a full linkage without hitting errors.
